components/provisioner/internal/operations/stages/deprovisioning/cleanup_cluster.go

```diff
@@ -0,0 +1,62 @@
+package deprovisioning
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/provisioner/internal/installation"
+	"github.com/kyma-incubator/compass/components/provisioner/internal/model"
+	"github.com/kyma-incubator/compass/components/provisioner/internal/operations"
+	"github.com/kyma-incubator/compass/components/provisioner/internal/util/k8s"
+	"github.com/sirupsen/logrus"
+)
+
+type CleanupClusterStep struct {
+	resourceClient installation.ResourceClient
+	nextStep       model.OperationStage
+	timeLimit      time.Duration
+}
+
+func NewCleanupClusterStep(nextStep model.OperationStage, timeLimit time.Duration) *CleanupClusterStep {
+	return &CleanupClusterStep{
+		nextStep:  nextStep,
+		timeLimit: timeLimit,
+	}
+}
+
+func (s *CleanupClusterStep) Name() model.OperationStage {
+	return model.CleanupCluster
+}
+
+func (s *CleanupClusterStep) TimeLimit() time.Duration {
+	return s.timeLimit
+}
+
+func (s *CleanupClusterStep) Run(cluster model.Cluster, _ model.Operation, logger logrus.FieldLogger) (operations.StageResult, error) {
+	logger.Infof("Starting cleanup cluster step for %s ...", cluster.ID)
```

Comment:
Logging errors in stage is redundant since it is done in Executor. As it comes to logging info consider either changing to Debug or remove.